buildSrc/src/main/kotlin/adapters.base-conventions.gradle.kts

```diff
@@ -0,0 +1,25 @@
+
+plugins {
+    `java-library`
+}
+
+java {
+    // using the toolchain to set the java version resulted in version errors in the cursed spigot setup.
+    sourceCompatibility = JavaVersion.VERSION_1_8
+    targetCompatibility = JavaVersion.VERSION_1_8
```

Comment:
Maybe it's related to the comment from the line above but why don't we define those values as 17 instead of java 8 ?